units/src/fee.rs

```diff
@@ -143,27 +144,47 @@ impl FeeRate {
         // No `?` operator in const context.
         match self.to_sat_per_kwu().checked_mul(weight.to_wu()) {
             Some(mul_res) => match mul_res.checked_add(999) {
-                Some(add_res) => Some(Amount::from_sat(add_res / 1000)),
+                // Unchecked because div by 1000 cannot be out of range.
+                Some(add_res) => Some(Amount::from_sat_unchecked(add_res / 1000)),
                 None => None,
             },
             None => None,
         }
     }
 }
 
-/// Computes the ceiling so that the fee computation is conservative.
 impl ops::Mul<FeeRate> for Weight {
     type Output = Amount;
 
+    /// Enables `fee = fee_rate * weight`.
+    ///
+    /// Computes the ceiling so that the fee computation is conservative. Consider using
+    /// [`FeeRate::checked_mul_by_weight`] if you do not control both the fee rate and the weight
+    /// because this function can panic.
+    ///
+    /// # Panics
+    ///
+    /// If `fee_rate * weight` exceeds `Amount::MAX_MONEY`.
```

Comment:
Please no. The doc on trait is more visible than on method.

Reply:
I'm not sure what you mean. I moved the docs on one of the `Mul` impls to be above the `impl` line and built them - both docs then look exactly the same.

Reply:
WTF, that must be a recent, very non-sensical change to rustdoc. Why would it put a doc from one place to another?

Anyway, the point is, if you have the doc on method people don't notice there's something significant about that impl as easily because the trait definition is collapsed by default.

Reply:
Also, not to hold this up, so maybe we can create an issue to minimize the use of `MAX_MONEY`.  I know it's taken from core and so it could just be a global const we add with the other core consts that are maintained.  Beyond that, I don't see much value in perpetuating the (somewhat awful const name imo) `MAX_MONEY` and prefer the more idiomatic `Amount::MAX`.

Reply:
@yancyribbens while I agree with you, your comment is unrelated to this discussion.

Reply:
> @yancyribbens while I agree with you, your comment is unrelated to this discussion

Apologies, I wasn't trying to de-rail your point, just that I had issue with the same line.

Reply:
And that issue is that `MAX_MONEY` is used doc comments which I don't think it should be.

Reply:
Oh, I see. I think you can click on the same line and start a new thread. Anyway, you're right.